### ISSUE TITLE
How should users structure plugin modules for custom visualization and analysis tools?

### ISSUE THREAD

**Issue by rachel3834** [TOMToolkit/tom_base#630]
**Is your feature request related to a problem? Please describe.**
Users often wish to develop customized plug-in modules for data visualization or analysis tools, for example plots for specific science use-cases, or analysis+plotting tools, such as a periodogram analysis.  These may often require interactive plotting and tabulation libraries, such as Plotly, Dash or custom Vue or React components. 

While users have a clear idea of the desired tool they wish to build, it is often unclear to them how to construct a plugin module, especially with reactive components.  

This relates to a request from @Markus Hundertmark and @Alex Razim, who can provide an example use-case.  

**Describe the solution you'd like**
Using Markus and Alex's use-case as an example, research the most effective way to integrate custom tools with the TOM Toolkit, and document the process so that users can follow it.  
Work with Markus and Alex as they develop their desired tools.  This may involve advising how best to structure their datasets within the TOM for easy of processing and retrieval.

**Comment by jchate6** [comment]
In most cases, this is likely a documentation issue. A few examples in the docs might go a long way.

**Comment by rachel3834** [comment]
Yes, absolutely.  A lot of users are not comfortable enough with Django to
know how to package their code, especially if it has dynamic components.

**Comment by jchate6** [comment]
I also want to link [tom_alerts_dash](https://github.com/TOMToolkit/tom_alerts_dash) and [tom_dash](https://github.com/TOMToolkit/tom_dash) here.

**Comment by jchate6** [comment]
See Issue #745
